src/store/builder/index.ts

```diff
@@ -361,33 +361,42 @@ export function generateConnectionLayoutKeyFor(source: IBlock, target: IBlock):
   ]
 }
 
```

Comment:
FYI, in this file I've tried fixing the position of the newly added blocks on the canvas.
@object-required @safydy